### PR TITLE
Update Ignition, Container Linux Config transpiler, and Cloud-init

### PR DESCRIPTION
**Pull request by dghubble**
* Rewrite Ignition/Fuze docs to focus on Container Linux Configs (review me plz)
  * matchbox users should not worry about Ignition for the most part
* Update/vendor container-linux-config-transpiler to v0.2.2
* Update/vendor Ignition to match Container Linux Config transpiler
* Update/vendor coreos-cloudinit to v1.13.0

### CODE REVIEW

File: Documentation/ignition.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
-# Ignition
```
Comment (crawford) [inline Documentation/ignition.md]: Should the name of this document be renamed as well?

File: Documentation/ignition.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+A Container Linux Config is a YAML document which declares how Container Linux instances' disks should be provisioned on network boot and first-boot from disk. Users can declare disk paritions, write files (regular files, systemd units, networkd units, etc.), and configure users. See the Container Linux Config [spec](https://coreos.com/os/docs/latest/configuration.html).
 
-## Fuze configs
+### Ignition
 
-Ignition 2.0.0+ configs are versioned, *machine-friendly* JSON documents (which contain encoded file contents). Operators should write and maintain configs in a *human-friendly* format, such as CoreOS [fuze](https://github.com/coreos/fuze) configs. As of `matchbox` v0.4.0, Fuze configs are the primary way to use CoreOS Ignition.
+Container Linux Configs are validated and converted to *machine-friendly* Ignition config (JSON) by matchbox when serving booting machines. This means, matchbox users need only write Container Linux Configs. The Ignition executor in the Container Linux initamfs implements declarative configuration.
```
Comment (crawford) [inline Documentation/ignition.md]: That last sentence is odd.

Comment (crawford) [inline Documentation/ignition.md]: s/Ignition config/Ignition configs/

File: Documentation/ignition.md [hunk 3]
```diff
@@ -23,38 +23,13 @@ Fuze template files can be added in the `/var/lib/matchbox/ignition` directory o
...
-        foo bar
-```
-
-Support for the older Ignition v1 format has been dropped, so CoreOS machines must be **1010.1.0 or newer**. Read the upstream Ignition v1 to 2.0.0 [migration guide](https://coreos.com/ignition/docs/latest/migrating-configs.html) to understand the reasons behind schema changes.
+Profiles can include a Container Linux Config for provisioning machines. Specify the Container Linux Config in a [Profile](matchbox.md#profiles) with `ignition_id`. When PXE booting, use the kernel option `coreos.first_boot=1` and `coreos.config.url` to point to the `matchbox` [Ignition endpoint](api.md#ignition-config).
 
 ## Examples
 
-Here is an example Fuze template. This template will be rendered into a Fuze config (YAML), using group metadata, selectors, and query params as template variables. Finally, the Fuze config is served to client machines as Ignition JSON.
+Here is an example Container Linux Config template. Variables will be interpreted using group metadata, selectors, and query params. Matchbox will convert the config to Ignition to serve Container Linux machines.
```
Comment (crawford) [inline Documentation/ignition.md]: The example below is still an Ignition config.

File: Documentation/ignition.md (hunk 1 above)
Comment (dghubble) [inline Documentation/ignition.md]: Its moved from ignition.md to container-linux-config.md in the last commit. I figured it was easier to review if the change and move were separate.

File: Documentation/ignition.md (hunk 2 above)
Comment (dghubble) [inline Documentation/ignition.md]: Agreed, how do we usually talk about Ignition, the binary shipping with Container Linux?

Comment (crawford) [inline Documentation/ignition.md]: "Ignition, the provisioning utility shipped in Container Linux, will execute this declarative config to realize the desired configuration."

Comment (dghubble) [inline Documentation/ignition.md]: Thanks

File: Documentation/ignition.md (hunk 3 above)
Comment (dghubble) [inline Documentation/ignition.md]: I think Github is playing tricks. It should be a Container Linux Config followed by the Ignition config that would be served. If you click expand in Github, it expands to show the Ignition first.

Full View https://github.com/coreos/matchbox/blob/28c6da21b92d25bc49719bc2b034d6b07acf72db/Documentation/ignition.md

Comment (crawford) [inline Documentation/ignition.md]: Yep, you're right.

File: Documentation/ignition.md (hunk 2 above)
Comment (dghubble) [inline Documentation/ignition.md]: Fixed

**Pull request merged**